### PR TITLE
React Calculator Project - Milestone 5 finished

### PR DESCRIPTION
**Pull request by Flakster**
## This PR includes

* Transformation of  App component from a function to a class
* Addition of handleClick function to ButtonPanel component
* Attachment of the handleClick function to the onClick event on Button component
* Addition of the logic of the calculations

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,37 @@
...
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = ButtonName => {
+    this.setState(
+      prevState => calculate(prevState, ButtonName),
+    );
+  }
```
Comment (SpaYco) [inline src/components/App.js]: - [ ] `handleClick` should be implemented inside the constructor

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -1,39 +1,46 @@
 import React from 'react';
+import propTypes from 'prop-types';
 import Button from './Button';
 
-function ButtonPanel() {
+function ButtonPanel(props) {
+  const { clickHandler } = props;
   return (
```
Comment (SpaYco) [inline src/components/ButtonPanel.js:7]: * [ ] ButtonPanel should have `handleClick` function implemented ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/implementHandleClick.png))

```javascript
handleClick = (buttonName) => {
  this.props.clickHandler(buttonName);
};
```

File: src/components/ButtonPanel.js [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
+        <Button buttonName="+" handleClick={clickHandler} />
       </div>
       <div className="button-row">
-        <Button buttonName="0" color="grey" wide="true" />
-        <Button buttonName="." color="grey" />
-        <Button buttonName="=" />
+        <Button buttonName="0" color="grey" wide handleClick={clickHandler} />
+        <Button buttonName="." color="grey" handleClick={clickHandler} />
+        <Button buttonName="=" handleClick={clickHandler} />
       </div>
```
Comment (SpaYco) [inline src/components/ButtonPanel.js:37]: - [ ] passe `handleClick` as `clickHandler` props to each `Button` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/passToButton.png))

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] check below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by Flakster** [comment]
# This new submission includes
* Bind the handleClick function in the App class Constructor
* clickHandler function,  which takes buttonName as an argument was implemented
* handleClick function is passed as a property to each button

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 

Great job  @Flakster, and thanks for making the required changes :hammer: :wrench:.

Now you can merge, your project is approved :checkered_flag: 

## Happy Coding :tophat:

**Pull request merged**